x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamOutput.java

```diff
@@ -68,6 +69,8 @@ public final class PlanStreamOutput extends StreamOutput implements org.elastics
      */
     protected final Map<EsField, Integer> cachedEsFields = new IdentityHashMap<>();
 
+    protected final Map<String, Integer> stringCache = new HashMap<>();
```

Comment:
Make sure this cache and the one above it are cleaned up through in the stream close() method.

Reply:
This is an interesting point.
We have other caches in the stream (eg. blocks cache) that could also be pretty large. Should we clean all of them? @nik9000 WDYT?
My assumption was that when we close a `StreamInput`/`Output` we also release it, so it would be garbage collected together with the caches, but maybe it's worth cleaning it up anyway.